src/api/common/createApi.ts

```diff
@@ -0,0 +1,23 @@
+import axios, { AxiosInstance, AxiosResponse, AxiosRequestConfig } from 'axios'
+
+import { CancelableRequest } from './types'
+
+export class ApiInstance {
+  client: AxiosInstance
+
+  constructor(config?: AxiosRequestConfig) {
+    const client = axios.create(config)
+
+    this.client = client
+  }
+
+  get<T = any, R = AxiosResponse<T>>(url: string, config?: AxiosRequestConfig): CancelableRequest<R> {
+    return (signal?: AbortSignal) =>
+      this.client.get(url, {
+        ...config,
+        signal,
+      })
+  }
```

Comment:
It's better to avoid long lines in code:
```suggestion
  get<T = any, R = AxiosResponse<T>>(
    url: string,
    config?: AxiosRequestConfig,
  ): CancelableRequest<R> {
    return (signal?: AbortSignal) =>
      this.client.get(url, {
        ...config,
        signal,
      })
  }
```

Reply:
Thanks, I've added the prettier config and fixed each file so that there are no long lines ✅